src/bfffxServer.spec.ts

```diff
@@ -234,24 +235,31 @@ describe('BFFFxServer', () => {
         .send();
 
       // redacted errors get logged
-      expect(consoleErrorMock).toBeCalledTimes(1);
-      const call = consoleErrorMock.mock.calls[0][0];
-      // is a JSON document
-      const parsedLog: any = JSON.parse(call);
-      // log tags include request info
-      expect(parsedLog.logTags?.method).toEqual('GET');
-      expect(parsedLog.logTags?.path).toEqual('/desktop/v1/recent-saves');
-      expect(parsedLog.logTags?.query.includes(CONSUMER_KEY)).toBeTruthy();
-      expect(parsedLog.logTags?.query.includes('count')).toBeTruthy();
-      expect(parsedLog.logTags?.user?.id).toEqual(cookies.a95b4b6);
-      expect(parsedLog.message).toEqual('I might reveal library misuse');
-      expect(parsedLog.originalError?.message).toEqual(
-        'I might reveal library misuse'
-      );
-      // error class
-      expect(parsedLog.originalError?.name).toEqual('Error');
-      // includes stack trace
-      expect(parsedLog.originalError?.stack).toBeTruthy();
+      expect(serverLoggerErrorMock).toBeCalledTimes(1);
+      // using ts-ignore as call could be undefined when no errors
+      // TypeScript does not like ts-ignore delcared top of the file as it interferes with global eslint rules
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      const call = serverLoggerErrorMock.mock?.calls[0][1];
+      // // log tags include request info
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      expect(call.tags?.method).toEqual('GET');
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      expect(call.tags?.path).toEqual('/desktop/v1/recent-saves');
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      expect(call.tags?.query.includes(CONSUMER_KEY)).toBeTruthy();
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      expect(call.tags?.query.includes('count')).toBeTruthy();
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      expect(call.tags?.user?.id).toEqual(cookies.a95b4b6);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      expect(call.error).toEqual('I might reveal library misuse');
```

Comment:
For the above ts ignore comments, what rule are we trying to suppress? Can we add that to our global lint rules?

Reply:
TS is complaning that `call` might be `undefined`, which it might be if no errors (also added this explanation to the top of the code chunk). Not positive if its a good idea to define this rule globally -- this is just a test case

Reply:
ahh I see. Can we try adding a check if `call` is not undefined and move all the asserts inside that if block? 

I think that will make the lint errors go away 🤔

Reply:
@Herraj TS still complains, I think it is because the meta in the `serverLogger` obj is `any` and TS is not happy when trying to access specific meta in the test like `tags`. It is still asking me to supress the error by applying `ts-ignore` in the test case